dio/lib/src/dio_mixin.dart

```diff
@@ -41,7 +41,8 @@ abstract class DioMixin implements Dio {
   /// The default [Transformer] that transfers requests and responses
   /// into corresponding content to send.
   @override
-  Transformer transformer = BackgroundTransformer();
+  Transformer transformer =
+      FusedTransformer(contentLengthIsolateThreshold: 50 * 1024);
```

Comment:
Can you add a comment explaining this specific number? Is it chosen rather arbitrarily or an educated guess?

Reply:
Does it make sense to be configurable?

Reply:
50 * 1024 is the hardcoded default in BackgroundTransformer, I chose the same to keep the behavior as close as possible to the old one. 
I did not measure how long it takes to decode on smartphones.

IMO it makes sense to make it configurable, for example, when running the whole HTTP client in a background isolate, users might not care about blocking the isolate for a few milliseconds, so they could set a higher (or -1 to disable spawning new isolates completely).